src/domain/submission/services/exporter/file-generators/article.ts

```diff
@@ -0,0 +1,354 @@
+import * as xmlbuilder from 'xmlbuilder';
+import axios from 'axios';
+import Submission, { ArticleType } from '../../models/submission';
+import user_api_url from '../../../../../config';
+
+interface Editor {
+    id: string;
+    email: string;
+    name: {
+        surname: string;
+        'given-names': string;
+    };
+    affiliations: string;
+}
+
+interface Affiliation {
+    name: string[];
+}
+
+interface ContinuumPerson {
+    id: string;
+    name: {
+        surname: string;
+        givenNames: string;
+    };
+    emailAddresses: { value: string }[];
+    affiliations?: Affiliation[];
+}
+
+const articleTypeMap: { [key in ArticleType]: number } = {
+    'research-article': 5,
+    'short-report': 13,
+    'tools-resources': 18,
+    'scientific-correspondence': 20,
+    'research-advance': 999, // @todo whats the value?
+    feature: 23,
+};
+
+const majorSubjectAreas: { [key: string]: string } = {
+    'biochemistry-chemical-biology': 'Biochemistry and Chemical Biology',
+    'cancer-biology': 'Cancer Biology',
+    'cell-biology': 'Cell Biology',
+    'chromosomes-gene-expression': 'Chromosomes and Gene Expression',
+    'computational-systems-biology': 'Computational and Systems Biology',
+    'developmental-biology': 'Developmental Biology',
+    ecology: 'Ecology',
+    'epidemiology-global-health': 'Epidemiology and Global Health',
+    'evolutionary-biology': 'Evolutionary Biology',
+    'genetics-genomics': 'Genetics and Genomics',
+    'human-biology-medicine': 'Human Biology and Medicine',
+    'immunology-inflammation': 'Immunology and Inflammation',
+    'microbiology-infectious-disease': 'Microbiology and Infectious Disease',
+    neuroscience: 'Neuroscience',
+    'physics-living-systems': 'Physics of Living Systems',
+    'plant-biology': 'Plant Biology',
+    'stem-cells-and-regenerative-medicine': 'Stem Cells and Regenerative Medicine',
+    'structural-biology-molecular-biophysics': 'Structural Biology and Molecular Biophysics',
+};
+
+// @todo: this should go in test setup
+const existingNames = [{ id: 1, first: 'J. Edward', last: 'Reviewer' }];
+
+// @todo: replace with ejp_name lookup
+const splitName = (name: string): { surname: string; givenNames: string } => {
+    const existing = existingNames.find(n => `${n.first} ${n.last}` === name);
+
+    if (existing) {
+        return {
+            surname: existing.last,
+            givenNames: existing.first,
+        };
+    }
+
+    const surname = name.substring(name.indexOf(' ') + 1);
+    const givenNames = name.substring(0, name.length - surname.length - 1);
+
+    return { surname, givenNames };
+};
+
+export default class ArticleGenerator {
+    private editors: Editor[] = [];
+    private affiliations: string[] = [];
+
+    constructor(private readonly submission: Submission) {}
+
+    async getEditor(id: string): Promise<Editor | null> {
+        const { data } = await axios.get<ContinuumPerson>(`${user_api_url}/editors/${id}`);
```

Comment:
Is there a PR for continuum-adaptor to implement this REST endpoint?

Reply:
wasn't this done as part of https://github.com/libero/reviewer/issues/595 ?

Reply:
That endpoint will get a list of editors given a role, not an individual - worth rasing another issue for this I think - I don't mind doing it while its fresh in my mind.

Reply:
ok i'll raise a separate issue